test/suite/test_util18.py

```diff
@@ -0,0 +1,192 @@
+    def populate(self):
+        cursor = self.session.open_cursor(self.uri, None)
+        for i in range(0, self.nentries):
+            key = 'KEY' + str(i)
+            val = 'VAL' + str(i)
+            cursor[key] = val
+        cursor.close()
+
+    # Check the given printlog file reflects the data written by 'populate'.
+    def check_populated_printlog(self, log_file, expect_keyval, expect_keyval_hex):
+        for i in range(0, self.nentries):
+            key = 'KEY' + str(i)
+            val = 'VAL' + str(i)
+            # Check if the KEY/VAL commits exist in the log file.
+            if expect_keyval:
+                self.check_file_contains(log_file, '"key": "%s\\u0000"' % key)
+                self.check_file_contains(log_file, '"value": "%s\\u0000"' % val)
+            else:
+                self.check_file_not_contains(log_file, '"key": "%s\\u0000"' % key)
+                self.check_file_not_contains(log_file, '"value": "%s\\u0000"' % val)
+
+            # Convert our KEY/VAL strings to their expected hex value.
+            hex_key = codecs.encode(key.encode(), 'hex')
+            val_key = codecs.encode(val.encode(), 'hex')
+            # Check if the KEY/VAL commits exist in the log file (in hex form).
+            if expect_keyval_hex:
+                self.check_file_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+            else:
+                self.check_file_not_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_not_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+
+    def test_printlog_file(self):
+        """
+        Run printlog on a populated table.
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        wt_args = ["printlog"]
+        # Append "-u" if we expect printlog to print user data.
+        if self.print_user_data:
+            wt_args.append("-u")
+        self.runWt(wt_args, outfilename='printlog.out')
+        self.check_non_empty_file('printlog.out')
+        self.check_populated_printlog('printlog.out', True & self.print_user_data, False)
+
+    def test_printlog_hex_file(self):
+        """
+        Run printlog with hexadecimal formatting on a populated table.
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        wt_args = ["printlog", "-x"]
+        # Append "-u" if we expect printlog to print user data.
+        if self.print_user_data:
+            wt_args.append("-u")
+        self.runWt(wt_args, outfilename='printlog-hex.out')
+        self.check_non_empty_file('printlog-hex.out')
+        self.check_populated_printlog('printlog-hex.out', True & self.print_user_data, True & self.print_user_data)
```

Comment:
Is the `True &` needed? I think you can pass in `self.print_user_data` as they are just `True` or `False` already.

Reply:
Yep, this was completely an over thought on my part. Fixed!